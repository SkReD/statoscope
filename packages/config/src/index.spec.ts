import { requireConfig } from './';

describe('requireConfig', () => {
  test('should work', () => {
    let resolved = requireConfig();
    resolved.path = String(resolved.path).replace(process.cwd(), '<pwd>');
    expect(resolved).toMatchSnapshot();

    resolved = requireConfig('fooooo');
    resolved.path = String(resolved.path).replace(process.cwd(), '<pwd>');
    expect(resolved).toMatchSnapshot();

    resolved = requireConfig(
      require.resolve('../../../test/fixtures/cli/validate/config'),
    );
    resolved.path = String(resolved.path).replace(process.cwd(), '<pwd>');
    expect(resolved).toMatchSnapshot();
  });
});
