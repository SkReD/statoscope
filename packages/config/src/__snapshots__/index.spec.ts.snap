// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`requireConfig should work 1`] = `
{
  "config": {},
  "path": "<pwd>/statoscope.config.js",
}
`;

exports[`requireConfig should work 2`] = `
{
  "config": {},
  "path": "<pwd>/fooooo",
}
`;

exports[`requireConfig should work 3`] = `
{
  "config": {
    "validate": {
      "plugins": [
        [
          "../../../../packages/stats-validator-plugin-webpack",
          "webpack",
        ],
      ],
      "rules": {
        "webpack/restricted-modules": [
          "error",
          [
            /\\\\/src\\\\//,
          ],
        ],
        "webpack/restricted-packages": [
          "error",
          [
            "foo",
          ],
        ],
      },
    },
  },
  "path": "<pwd>/test/fixtures/cli/validate/config.js",
}
`;
